### ISSUE TITLE
[Bug]:  when match search varchar type column throw ERROR 3013

### ISSUE THREAD

**Issue by XiaozanZhang** [infiniflow/infinity#1265]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Version or Commit ID

v0.2.0

### Other environment information

```Markdown
centos7.6, 256G memory
```

### Actual behavior and How to reproduce it

table fileds = {id: int , embedding: vector (1024*float), content : varchar}
index={embedding: hnsw M:16 ef:50 metric:ip, content: Fulltext}

search:
res = table.output([""]).match("content", "DS-K3AJ303/Dm140", 'topn=3').to result()

return error:
Exception ERROR:3013 Invalid query statemet : OrQueryNode should not have both not child and non-not child

### Expected behavior

return result and not exception

**Comment by yingfeng** [comment]
Hi, currently '-' is used for NOT query specified within the search grammer. You could use quotation marks to specify do "DS-K3AJ303/Dm140" as a phrase query so that the token inside the quotes is escaped.

**Comment by yingfeng** [comment]
Just adjust through following query statement:
```
 res = table.output([""]).match("content", "\"DS-K3AJ303/Dm140\"", 'topn=3').to result()
```

**Comment by yingfeng** [comment]
Converting the query into phrase query is a workaround. After the feature #1266 has been supported, the following query statement should work at that time.

```
 res = table.output([""]).match("content", "DS\-K3AJ303/Dm140", 'topn=3').to result()
```

**Comment by yingfeng** [comment]
Fixed by #1303

Special characters in query should be escaped by users.
